### DISCUSSION TITLE
0-Interval calendar alarms

### DISCUSSION THREAD

**Discussion by snugghash** [tutao/tutanota#3890] [Ideas]
Personally I use my encrypted calendar to schedule my work items too. And often, I'd like to get a notification/reminder _when_ the event begins, and not just (>)5 min before the event. 

Would the way to add that feature be as simple as adding another line at https://github.com/tutao/tutanota/blob/7ca87fcf2faa2b4f6c5687f89385142e7af7a735/src/calendar/view/CalendarEventEditDialog.ts#L66 and at https://github.com/tutao/tutanota/blob/7ca87fcf2faa2b4f6c5687f89385142e7af7a735/src/api/common/TutanotaConstants.ts#L447, for `0M`? Does the underlying engine support that?

**Comment by snugghash** [comment]
I could also directly make a pull request and get some feedback that way - can someone confirm if I'm on the right track?

**Comment by charlag** [comment]
This is not as simple as that. We need to release mobile and desktop apps with support for the new trigger first and ideally we should rewrite them to parse alarm trigger dynamically.

**Comment by snugghash** [comment]
Understandable - how much work do you estimate that would be, and is it worth it for other people/projects as well?

**Comment by charlag** [comment]
It is hard to do for most people as you need to be able to build iOS/Android/Desktop apps. Programming-wise it is not terribly difficult. I would say that it would be easier for us to implement this specific issue.

**Reply by snugghash** [reply]
Can I perhaps ask how much of a priority this is? Alternatively I can spend a few hours on this - I've worked on all three  platforms before, although I've yet to set up my own dev env for Tutanota.

**Reply by charlag** [reply]
I can say that this is not a topmost priority.
If you are willing to write a simple parser for the trigger, you are more than welcome to.
